modules/social_features/social_group/modules/social_group_request/social_group_request.module

```diff
@@ -0,0 +1,383 @@
+    }
+  }
+
+  $account = \Drupal::currentUser();
+  if ($account->isAnonymous()) {
+    $variables['anonymous_request'] = TRUE;
+    $variables['group_operations_url'] = Url::fromRoute('social_group_request.anonymous_request_membership', ['group' => $group->id()]);
+    $variables['#attached']['library'][] = 'core/drupal.dialog.ajax';
+    $variables['#attached']['library'][] = 'social_group_request/social_group_popup';
+    return;
+  }
+
+  if (
+    !$group->hasPermission('request group membership', $account) ||
+    !$group->hasField('allow_request')
+  ) {
+    $variables['allow_request'] = FALSE;
+    return;
+  }
+
+  $variables['allow_request'] = TRUE;
+  $variables['group_operations_url'] = Url::fromRoute('grequest.request_membership', ['group' => $group->id()]);
+
+  $contentTypeConfigId = $group
+    ->getGroupType()
+    ->getContentPlugin('group_membership_request')
+    ->getContentTypeConfigId();
+
+  $request = \Drupal::entityQuery('group_content')
+    ->condition('type', $contentTypeConfigId)
+    ->condition('gid', $group->id())
+    ->condition('entity_id', $account->id())
+    ->condition('grequest_status', GroupMembershipRequest::REQUEST_PENDING)
+    ->count()
+    ->execute();
+
+  if ($request > 0) {
+    $variables['requested'] = TRUE;
+    $variables['group_operations_url'] = Url::fromRoute('social_group_request.cancel_request', ['group' => $group->id()]);
+  }
+
+  $variables['#attached']['library'][] = 'social_group_request/social_group_popup';
+  $variables['#attached']['library'][] = 'social_group_request/social_group_request_popup';
+  $variables['#attached']['library'][] = 'core/drupal.dialog.ajax';
+  $variables['#cache']['tags'][] = 'request-membership:' . $group->id();
+}
+
+/**
+ * Implements hook_activity_send_email_notifications_alter().
+ */
+function social_group_request_activity_send_email_notifications_alter(array &$items, array $email_message_templates) {
+  if (isset($email_message_templates['request_to_join_a_group'])) {
+    $items['what_manage']['templates'][] = 'request_to_join_a_group';
+  }
+
+  if (isset($email_message_templates['approve_request_join_group'])) {
+    $items['message_to_me']['templates'][] = 'approve_request_join_group';
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function social_group_request_form_alter(array &$form, FormStateInterface $form_state, $form_id) {
+  $social_group_types = [
+    'open_group',
+    'closed_group',
+    'public_group',
+  ];
+
+  \Drupal::moduleHandler()->alter('social_group_types', $social_group_types);
+
+  $group_membership_add_forms = [];
+  foreach ($social_group_types as $social_group_type) {
+    $group_membership_add_forms[] = "group_content_{$social_group_type}-group_membership_add_form";
+  }
+
+  $route = \Drupal::routeMatch()->getRouteName();
+  if (in_array($form_id, $group_membership_add_forms) && $route === 'grequest.group_request_membership_approve') {
+    if (isset($form['entity_id']['widget'][0]['target_id'])) {
```

Comment:
Why are we doing this? Could you leave some inline comments?

Reply:
We're removing possibility to select members to add to group. Some inline comments are already added